### PR TITLE
added keep start metadata option

### PR DESCRIPTION
**Pull request by jrmlhermitte**
Not finished yet (would need a test, but I did test this myself and works as I intend it), but just to start discussion. I think it would be good to have an option to keep propagating metadata from start docs. What do you think?

### CODE REVIEW

File: shed/event_streams.py [hunk 1]
```diff
@@ -381,7 +385,11 @@ def start(self, docs, clear=True):
             self._clear()
         self.run_start_uid = str(uuid.uuid4())
         self.parent_uids = [doc['uid'] for doc in docs if doc]
-        new_start_doc = dict(uid=self.run_start_uid,
+        new_start_doc = dict()
+        if self.keep_start:
+            for doc in docs:
```
Comment (CJ-Wright) [inline shed/event_streams.py]: Note that this has right to left resolution, rather than left to right.

**Comment by ordirules** [comment]
thanks. sounds like it may be best to stick to using `Eventify` as best as possible rather than adding this convenience until other features have been resolved. So I will perhaps  close this unless you have an objection?

File: shed/event_streams.py (hunk 1 above)
Comment (ordirules) [reply shed/event_streams.py]: thanks, noted. it should conform with how `doc_contents` resolution order to be consisten

**Comment by jrmlhermitte** [comment]
closed. We are still interested in propagating metadata but this can be done other ways (like splitting into two streams, one being eventify and merging if needed).

**Pull request closed without merging**